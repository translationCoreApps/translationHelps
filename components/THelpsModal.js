/**
  * @author      Manny Colon
  * @description This component displays a modal when the user clicks the
  * new-window glyphicon button on translationHelps component.
******************************************************************************/
import React from 'react';
import {Modal, Glyphicon} from 'react-bootstrap';
import style from '../css/style';

class THelpsModal extends React.Component {
  render() {
    let { onHide, children } = this.props;
    var page = document.getElementById("modalbody");
    if (page) {
      page.scrollTop = 0;
    }

    return (
      <Modal {...this.props} bsSize="lg" aria-labelledby="contained-modal-title-sm">
        <Modal.Header style={{backgroundColor: "var(--background-color-light)", borderColor: "var(--text-color)"}}>
          <Modal.Title id="contained-modal-title-sm"
                      style={style.modalTitle}>
              translationHelps
              <Glyphicon
                  onClick={onHide}
                  glyph={"remove"}
                  style={{color: "var(--text-color)", cursor: "pointer", fontSize: "18px", float: "right"}}
              />
          </Modal.Title>
        </Modal.Header>
          <Modal.Body id="modalbody" style={style.tHModalContent}>
            {children}
          </Modal.Body>
        <Modal.Footer style={{padding: '0', backgroundColor: "var(--background-color-light)", borderColor: "var(--text-color)"}}>
          <button className="btn-prime" onClick={() => onHide()}>Close</button>
        </Modal.Footer>
      </Modal>
    );
  }
}

export default THelpsModal;
