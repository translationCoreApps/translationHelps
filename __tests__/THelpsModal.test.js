/* eslint-env jest */
import React from 'react';
import THelpsModal from '../src/components/THelpsModal';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const props = {
  translate:k=>k,
  onHide: jest.fn(),
  children: "[children]",
  show: true
};

describe('THelpsModal component Tests', () => {
  it('Check THelpsModal component', () => {
    const wrapper = shallow(<THelpsModal {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});