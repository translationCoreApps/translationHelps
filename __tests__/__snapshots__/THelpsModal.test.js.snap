// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THelpsModal component Tests Check THelpsModal component 1`] = `
<Modal
  animation={true}
  aria-labelledby="contained-modal-title-sm"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  bsSize="lg"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
  translate={[Function]}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
    style={
      Object {
        "backgroundColor": "var(--background-color-light)",
        "borderColor": "var(--text-color)",
      }
    }
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
      id="contained-modal-title-sm"
      style={
        Object {
          "color": "var(--text-color-dark)",
          "textAlign": "center",
        }
      }
    >
      translationHelps
      <Glyphicon
        bsClass="glyphicon"
        glyph="remove"
        onClick={[Function]}
        style={
          Object {
            "color": "var(--text-color)",
            "cursor": "pointer",
            "float": "right",
            "fontSize": "18px",
          }
        }
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
    id="modalbody"
    style={
      Object {
        "backgroundColor": "var(--background-color-light)",
        "color": "var(--text-color-dark)",
        "fontSize": "18px",
        "height": "500px",
        "overflowY": "auto",
        "padding": "45px 80px",
      }
    }
  >
    [children]
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
    style={
      Object {
        "backgroundColor": "var(--background-color-light)",
        "borderColor": "var(--text-color)",
        "padding": "0",
      }
    }
  >
    <button
      className="btn-prime"
      onClick={[Function]}
    >
      Close
    </button>
  </ModalFooter>
</Modal>
`;
